src/drive/mobile/containers/FirstUploadModal.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react'
+import Modal, { ModalContent } from 'cozy-ui/react/Modal'
+import { connect } from 'react-redux'
+import withPersistentState from '../lib/withPersistentState'
+import styles from '../styles/uploadprogression'
+import { isIos } from '../lib/device'
+
+const MINIMUM_LONG_UPLOAD_FILES_COUNT = 50
+
+class FirstUploadModal extends Component {
+  render() {
+    const { onClose } = this.props
+    const { t } = this.context
+
+    return (
+      <Modal title={t('mobile.first_sync.title')} withCross={false}>
+        <ModalContent>
+          <h4>{t('mobile.first_sync.tips')}</h4>
+          <ul>
+            <li>{t('mobile.first_sync.tip_wifi')}</li>
+            <li>{t('mobile.first_sync.tip_bed')}</li>
+            {isIos() && <li>{t('mobile.first_sync.tip_lock')}</li>}
+          </ul>
+          <p>{t('mobile.first_sync.result')}</p>
+          <button onClick={onClose} className={styles['btn--full-width']}>
+            {t('mobile.first_sync.button')}
+          </button>
+        </ModalContent>
+      </Modal>
+    )
+  }
+}
+
+const showOnlyOnce = WrappedComponent => {
+  return class extends Component {
+    state = {
+      shown: false
+    }
+
+    markAsShown = () => {
+      this.setState({ shown: true })
+    }
+
+    render() {
+      const { uploading } = this.props
+      const { shown } = this.state
+
+      if (uploading && !shown)
+        return <WrappedComponent {...this.props} onClose={this.markAsShown} />
```

Comment:
Maybe we should rename the `shown` state to `viewed` or `read` like in _mark as read_ or _mark as viewed_. I've never seen _mark as shown_, don't we?

Reply:
Well you're right, `viewed` is probably more usual.